validation/src/main/scala/hmda/validation/rules/lar/quality/Q022.scala

```diff
@@ -0,0 +1,28 @@
+package hmda.validation.rules.lar.quality
+
+import hmda.model.fi.lar.LoanApplicationRegister
+import hmda.validation.dsl.Result
+import hmda.validation.dsl.PredicateCommon._
+import hmda.validation.dsl.PredicateSyntax._
+
+import scala.concurrent.{ ExecutionContext, Future }
+import scala.util.Try
+
+object Q022 {
+  def apply(lar: LoanApplicationRegister, year: Int): Result = {
+    // This edit should not fail if applicationDate is equal to "NA" or is otherwise non-convertible
+    val applicationYear = Try(lar.loan.applicationDate.substring(0, 4).toInt).getOrElse(year)
```

Comment:
It looks like the code to extract `applicationYear` is the same for both of the `apply` methods. You could extract that into a method to make this cleaner.